lib/validate.js

```diff
@@ -0,0 +1,157 @@
+const {
+  ERROR_BAD_PRIVATE,
+  ERROR_BAD_POINT,
+  ERROR_BAD_TWEAK,
+  throwError,
+  ERROR_BAD_HASH,
+  ERROR_BAD_EXTRA_DATA,
+  ERROR_BAD_SIGNATURE,
+} = require("./validate_wasm");
+
+const PRIVATE_KEY_SIZE = 32;
+const PUBLIC_KEY_COMPRESSED_SIZE = 33;
+const PUBLIC_KEY_UNCOMPRESSED_SIZE = 65;
+const TWEAK_SIZE = 32;
+const HASH_SIZE = 32;
+const EXTRA_DATA_SIZE = 32;
+const SIGNATURE_SIZE = 64;
+
+const BN32_ZERO = new Uint8Array(32);
+const BN32_N = new Uint8Array([
+  255,
+  255,
+  255,
+  255,
+  255,
+  255,
+  255,
+  255,
+  255,
+  255,
+  255,
+  255,
+  255,
+  255,
+  255,
+  254,
+  186,
+  174,
+  220,
+  230,
+  175,
+  72,
+  160,
+  59,
+  191,
+  210,
+  94,
+  140,
+  208,
+  54,
+  65,
+  65,
+]);
+
+function isUint8Array(value) {
+  return value instanceof Uint8Array;
```

Comment:
iirc there was an issue in another library where some environments this doesn't work well.

Reply:
I remember only this issue -- https://github.com/cryptocoinjs/secp256k1-node/issues/175 :thinking: I think that `value` can be anything while it's correct work with [Uint8Array.prototype.set](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/TypedArray/set) (we copy input value data to WASM input buffer with `set`).

Reply:
Do you think it worth to change to something like:
```js
function itemsInRange(value) {
  for (let i = 0; i < value.length; ++i) {
    if (value[i] < 0 || value[i] > 0xff) return false;
  }
  return true;
}

function isUint8ArrayLike(value, length) {
  return value instanceof Uint8Array || (
    value !== undefined && value !== null && value.length === length && itemsInRange(value)
  )
}
```